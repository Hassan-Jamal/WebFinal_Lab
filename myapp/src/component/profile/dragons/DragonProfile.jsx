import PropTypes from 'prop-types';

// import './dragonProfile.scss';

const DragonProfile = ({ dragon }) => (
  <li className="Dragon-reserved">
    {dragon.dragonName}
  </li>
);

DragonProfile.propTypes = {
  dragon: PropTypes.shape({
    dragonName: PropTypes.string,
    dragonDescription: PropTypes.string,
    reserved: PropTypes.bool,
    flickrImages: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.string,
  }).isRequired,
};

export default DragonProfile;
