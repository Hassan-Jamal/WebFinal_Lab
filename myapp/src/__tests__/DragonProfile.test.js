import React from 'react';
import renderer from 'react-test-renderer';
import DragonProfile from '../components/profile/dragon/DragonProfile';

describe('DragonProfile', () => {
  it('renders correctly', () => {
    const dragon = {
      dragonName: 'Falcon 9',
      dragonDescription: 'A two-stage Dragon...',
      reserved: true,
      flickrImages: ['url1', 'url2'],
      id: 'dragon1',
    };

    const dragonProfile = renderer.create(<DragonProfile dragon={dragon} />).toJSON();

    expect(dragonProfile).toMatchSnapshot();
  });
});
