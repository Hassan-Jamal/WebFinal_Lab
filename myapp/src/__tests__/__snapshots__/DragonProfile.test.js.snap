// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragonProfile renders correctly 1`] = `
<li
  className="dragon-reserved"
>
  Falcon 9
</li>
`;
